src/main/java/domain/Line.java

```diff
@@ -0,0 +1,54 @@
+package domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import utils.NumberGenerator;
+
+public class Line {
+
+    private static final int MIN_NUMBER_RETURN_TRUE = 4;
+
+    private final List<Point> points;
+
+    private Line(List<Point> points) {
+        this.points = new ArrayList<>(points);
+    }
+
+    public static Line create(int numberOfPeople, NumberGenerator numberGenerator) {
+        List<Point> points = new ArrayList<>();
+        int numberOfPointsToGenerate = numberOfPeople - 1;
+        for (int i = 0; i < numberOfPointsToGenerate; i++) {
+            addPoint(points, numberGenerator);
+        }
+        return new Line(points);
+    }
+
+    private static void addPoint(List<Point> points, NumberGenerator numberGenerator) {
+        if (points.isEmpty()) {
+            points.add(generatePoint(numberGenerator.generate()));
+            return;
+        }
+
+        if (isPreviousPointPassable(points)) {
+            points.add(Point.BLOCKED);
+            return;
+        }
+        points.add(generatePoint(numberGenerator.generate()));
+    }
+
+    private static Point generatePoint(int number) {
+        if (number >= MIN_NUMBER_RETURN_TRUE) {
+            return Point.PASSABLE;
+        }
+        return Point.BLOCKED;
+    }
```

Comment:
어떤 Point가 생성될지 판단하고 생성하는 책임을 Point enum에게 맡겨보면 어떨까요?
```suggestion
public enum Point {
    ...
    private static Point of(int number) {
        ...
    }
```

Reply:
확실히 Point 하나를 생성하는 책임은 Point 객체가 직접 가지고 있는게 더 자연스러운 것 같습니다. 
그런데 보여주신 예시 코드에서는 `of` 메소드가 `private` 이네용.. 혹시 제가 사용한 `public` 말고 `private` 으로 메소드 접근제어자를 지정하여 사용하는 방법이 있을까요?

Reply:
아 이건 제가 복붙해서 private 으로 들어가버렸네요.. public 이 맞습니다..!!